ui/site/src/component/topBar.ts

```diff
@@ -169,4 +169,21 @@ export default function () {
       })
       .bind('s', () => $input[0]!.focus());
   }
+
+  {
+    let lastScrollY = 0;
+    const header = document.getElementById('top')!;
+
+    window.addEventListener('scroll', () => {
```

Comment:
` window.addEventListener('scroll'` is very bad for performance, as the callback is allowed to return false to prevent the default behaviour. So our JS code must run before the browser can scroll the page.
I think we can use the `passive` setting to fix that